target/linux/ath79/modules.mk

```diff
@@ -14,3 +14,70 @@ define KernelPackage/leds-reset/description
 endef
 
 $(eval $(call KernelPackage,leds-reset))
+
+
+NETWORK_DEVICES_MENU:=Network Devices
+
+define KernelPackage/ag71xx
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Atheros AR7XXX/AR9XXX built-in ethernet mac support
+  KCONFIG:=CONFIG_AG71XX \
+	   CONFIG_AG71XX_DEBUG_FS=y
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/atheros/ag71xx/ag71xx.ko \
+	 $(LINUX_DIR)/drivers/net/ethernet/atheros/ag71xx/ag71xx_mdio.ko
+  AUTOLOAD:=$(call AutoLoad,20,ag71xx ag71xx_mdio,1)
+endef
+
+define KernelPackage/ag71xx/description
+  Atheros AR7XXX/AR9XXX built-in ethernet mac support
+endef
+
+$(eval $(call KernelPackage,ag71xx))
+
+
+define KernelPackage/phy-at803x
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Qualcomm Atheros 8337 internal PHY
+  KCONFIG:=CONFIG_AT803X_PHY
+  FILES:=$(LINUX_DIR)/drivers/net/phy/at803x.ko
+  AUTOLOAD:=$(call AutoLoad,19,at803x,1)
+endef
+
+define KernelPackage/phy-at803x/description
+  Qualcomm Atheros 8337 internal PHY (conflicts with kmod-phy-ar8xxx)
+endef
+
+$(eval $(call KernelPackage,phy-at803x))
+
+
+define KernelPackage/phy-ar8xxx
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=AR8216/8327 Ethernet PHYs
+  DEPENDS:=+kmod-swconfig
+  KCONFIG:=CONFIG_AR8216_PHY \
+	   CONFIG_AR8216_PHY_LEDS=y
+  FILES:=$(LINUX_DIR)/drivers/net/phy/ar8xxx.ko
+  AUTOLOAD:=$(call AutoLoad,18,ar8xxx,1)
+endef
+
+define KernelPackage/phy-ar8xxx/description
+  Kernel modules for AR8216/8327 Ethernet PHYs (conflicts with kmod-phy-at803x)
+endef
+
+$(eval $(call KernelPackage,phy-ar8xxx))
+
+
+define KernelPackage/phy-ip17xx
```

Comment:
I saw there are already kernel packages for [`kmod-switch-ip17xx`](https://github.com/luizluca/openwrt/blob/bbc7de3b164f43f44feff9ef8e72a2b475953eec/package/kernel/linux/modules/netdevices.mk#L405) (used by one ramips device) and [`kmod-switch-ar8xxx`](https://github.com/luizluca/openwrt/blob/bbc7de3b164f43f44feff9ef8e72a2b475953eec/package/kernel/linux/modules/netdevices.mk#LL517C35-L517C35) (unused). Not sure how well it would work (might be a horrible idea), but, if you only need minor differences to those packages; example, change the `AutoLoad` priority, you might be able to save some duplication (at the expense of clarity) by calling those already-defined packages.
Probably best to wait for some feedback on this idea (calling existing packages) before you spend too much time on it.

Example:
I had to remove my `openwrt/tmp` folder, to get the build system to regenerate `tmp/.config-package.in` 
```
define KernelPackage/switch-ip17xx-18
  $(call KernelPackage/switch-ip17xx)
  CONFLICTS:=kmod-switch-ip17xx
  TITLE+= (autoload priority 18)
  AUTOLOAD:=$(call AutoLoad,18,ip17xx,1)
endef

define KernelPackage/switch-ip17xx-18/description
  $(call KernelPackage/switch-ip17xx/description)
endef

$(eval $(call KernelPackage,switch-ip17xx-18))
```

`CONFIG_AT803X_PHY` is widely used, so you may be able to put that in netdevices

Reply:
> I saw there are already kernel packages for [`kmod-switch-ip17xx`](https://github.com/luizluca/openwrt/blob/bbc7de3b164f43f44feff9ef8e72a2b475953eec/package/kernel/linux/modules/netdevices.mk#L405) (used by one ramips device) and [`kmod-switch-ar8xxx`](https://github.com/luizluca/openwrt/blob/bbc7de3b164f43f44feff9ef8e72a2b475953eec/package/kernel/linux/modules/netdevices.mk#LL517C35-L517C35) (unused). Not sure how well it would work (might be a horrible idea), but, if you only need minor differences to those packages; example, change the `AutoLoad` priority, you might be able to save some duplication (at the expense of clarity) by calling those already-defined packages. Probably best to wait for some feedback on this idea (calling existing packages) before you spend too much time on it.

How I missed those! Thanks! I though I did a grep looking for the symbol but I might have done something wrong. I'll move them to netdevices. I just "synced" the priority close to similar drivers and increased the ethernet driver even further.

> Example: I had to remove my `openwrt/tmp` folder, to get the build system to regenerate `tmp/.config-package.in`

I also noticed that I had to remove openwrt/tmp to get them in the menu. Is that expected? It looks like the build script is not keeping track of target/*/modules.mk
 
> `CONFIG_AT803X_PHY` is widely used, so you may be able to put that in netdevices

I'll migrate to netdevices.